x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/logging/DeprecationIndexingComponent.java

```diff
@@ -74,17 +86,59 @@ public DeprecationIndexingComponent(
             consumer
         );
         enableDeprecationLogIndexing(enableDeprecationLogIndexingDefault);
+
+    }
+
+    public static DeprecationIndexingComponent createDeprecationIndexingComponent(
+        Client client,
+        Settings settings,
+        RateLimitingFilter rateLimitingFilterForIndexing,
+        boolean enableDeprecationLogIndexingDefault,
+        ClusterService clusterService
+    ) {
+        final DeprecationIndexingComponent deprecationIndexingComponent = new DeprecationIndexingComponent(
+            client,
+            settings,
+            rateLimitingFilterForIndexing,
+            enableDeprecationLogIndexingDefault,
+            clusterService
+        );
+
+        clusterService.addListener(deprecationIndexingComponent);
+        clusterService.getClusterSettings()
+            .addSettingsUpdateConsumer(WRITE_DEPRECATION_LOGS_TO_INDEX, deprecationIndexingComponent::enableDeprecationLogIndexing);
+
+        return deprecationIndexingComponent;
+    }
+
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        if (flushEnabled.get() == false) {
+            if (event.metadataChanged()) {
+                final IndexLifecycleMetadata indexLifecycleMetadata = event.state().metadata().custom(IndexLifecycleMetadata.TYPE);
+
+                if (event.state().getMetadata().templatesV2().containsKey(".deprecation-indexing-template")
+                    && indexLifecycleMetadata != null
+                    && indexLifecycleMetadata.getPolicies().containsKey(".deprecation-indexing-ilm-policy")) {
+                    flushEnabled.set(true);
```

Comment:
Maybe also remove this listener if flushEnable is being set to true? (using `ClusterService#removeListener`)
That way there is no need to to keep checking whether flush has been enabled after ilm policy / template has been added.

Reply:
I missed the fact that `ClusterService#removeListener` was already being invoked, so this comment is not relevant and can be resolved